src/sidebar/build-thread.js

```diff
@@ -203,96 +198,93 @@ function sort(threads, compareFn) {
 }
 
 /**
- * Return a copy of `thread` with siblings of the top-level thread sorted according
- * to `compareFn` and replies sorted by `replyCompareFn`.
+ * Return a new `Thread` object with all (recursive) `children` arrays sorted.
+ * Sort the children of top-level threads using `compareFn` and all other
+ * children using `replyCompareFn`.
+ *
+ * @param {Thread} thread
+ * @param {(a: Annotation, b: Annotation) => boolean} compareFn - Less-than
+ *         comparison function for sorting top-level annotations
+ * @param {(a: Annotation, b: Annotation) => boolean} replyCompareFn - Less-than
+ *       comparison function for sorting replies
+ * @return {Thread}
  */
 function sortThread(thread, compareFn, replyCompareFn) {
-  const children = thread.children.map(function (child) {
-    return sortThread(child, replyCompareFn, replyCompareFn);
-  });
+  const children = thread.children.map(child =>
+    sortThread(child, replyCompareFn, replyCompareFn)
+  );
 
-  return Object.assign({}, thread, {
-    children: sort(children, compareFn),
-  });
+  return { ...thread, ...{ children: sort(children, compareFn) } };
 }
 
 /**
- * Return a copy of @p thread with the `replyCount` and `depth` properties
+ * Return a copy of thread with the `replyCount` and `depth` properties
  * updated.
+ *
+ * @param {Thread} thread
+ * @param {number} depth
+ * @return {Thread}
  */
 function countRepliesAndDepth(thread, depth) {
-  const children = thread.children.map(function (c) {
-    return countRepliesAndDepth(c, depth + 1);
-  });
-  return Object.assign({}, thread, {
-    children: children,
-    depth: depth,
-    replyCount: children.reduce(function (total, child) {
-      return total + 1 + child.replyCount;
-    }, 0),
-  });
+  const children = thread.children.map(c => countRepliesAndDepth(c, depth + 1));
+  const replyCount = children.reduce(
+    (total, child) => total + 1 + child.replyCount,
+    0
+  );
+  return {
+    ...thread,
+    ...{
+      children,
```

Comment:
Object spread is not needed here for the `children`, `depth` and `replyCount` properties.